tests/bounded.rs

```diff
@@ -1,17 +1,27 @@
 #![allow(clippy::bool_assert_comparison)]
 
+use std::future::Future;
 use std::sync::atomic::{AtomicUsize, Ordering};
+use std::task::{Context, Poll};
 use std::thread::sleep;
 use std::time::Duration;
 
 use async_channel::{bounded, RecvError, SendError, TryRecvError, TrySendError};
 use easy_parallel::Parallel;
 use futures_lite::{future, prelude::*};
+use waker_fn::waker_fn;
 
 fn ms(ms: u64) -> Duration {
     Duration::from_millis(ms)
 }
 
+fn poll_now<F: Future + Unpin>(f: &mut F) -> Poll<F::Output> {
```

Comment:
You could probably just use `poll_once` from `futures_lite` here.

Reply:
You are right, I discarded this solution because the future must be kept alive so can't be consumed, it didn't come to my mind that passing a mutable ref would work just as well since it implement `Future` too...